src/imperative/imperative.cc

```diff
@@ -233,7 +233,7 @@ void Imperative::RecordOp(
 
   nnvm::ObjectPtr node = nnvm::Node::Create();
   node->attrs = std::move(attrs);
-  node->attrs.name = "node_" + std::to_string(node_count_++);
+  node_count_ += 1;
```

Comment:
Why do you disable naming the ops in autograd recording?

Reply:
Because this naming will overwrite the name assigned previously to `node->attrs.name`, which is less descriptive.

Reply:
Can you reference the prior name assignment you have in mind?

Reply:
As an example:

```
C = mx.symbol.dot(A, B, name='dot')
```

Here the name for C would be `dot`, however, autograd will overwrite its name to be `node_0` (or something like that).

Reply:
The example you give relies on the NameManager feature, which is only available in symbolic API. Note that autograd and dc do not apply to the symbolic API, thus they will not overwrite names set by the NameManager. That's why the names are generated via `node_count_ += 1;`, which you now deleted